composer/loggers/mosaicml_logger.py

```diff
@@ -88,13 +90,30 @@ def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None) -> No
         self._log_metadata(metrics)
 
     def after_load(self, state: State, logger: Logger) -> None:
+        # Log model data downloaded and initialized for run events
+        self._log_metadata({'model_initialized_time': time.time()})
         # Log WandB run URL if it exists. Must run on after_load as WandB is setup on event init
         for callback in state.callbacks:
             if isinstance(callback, WandBLogger):
                 run_url = callback.run_url
                 if run_url is not None:
                     self._log_metadata({'wandb/run_url': run_url})
 
+    def fit_start(self, state: State, logger: Logger) -> None:
+        # Log max duration in either epochs or tokens for run events
+        if state.max_duration.unit == TimeUnit.TOKEN:
```

Comment:
Can we just log `max_duration_value` and `max_duration_units`?

Reply:
We're planning to log [epoch x/xx] [batch x/xx] if the user submits training duration in ep or ba, so I think we want to calculate num batches per epoch in addition to the max duration.

Reply:
I think you can draw inspiration from https://github.com/mosaicml/composer/blob/5dff73082ec5f7dbfe2c1ee7c711b6acf3207742/composer/loggers/console_logger.py#L185-L206. but basically you'll need to account for the different possibilities and do different things. for the finetuning api everything will always be defined, so its just to make this logger robust.

Reply:
You can look at https://github.com/mosaicml/composer/blob/dev/composer/callbacks/runtime_estimator.py#L99-L104 for inspiration

This function computes elapsed duration more robustly -- you can rip out parts of it https://github.com/mosaicml/composer/blob/dev/composer/callbacks/runtime_estimator.py#L90